vividus-plugin-email/src/main/java/org/vividus/bdd/email/steps/EmailSteps.java

```diff
@@ -0,0 +1,210 @@
+     * <li>REPLY_TO</li>
+     * </ul>
+     * Notes:
+     * <ul>
+     * <li>CONTAINS, MATCHES, IS_EQUAL_TO, DOES_NOT_CONTAIN are not allowed to be used with <i>date</i>
+     * parameters</li>
+     * <li><b>rule</b> names are case insensitive</li>
+     * <li>multiple <i>address</i> parameters can be checked by separating expected value by commas</li>
+     * <li><i>date</i> parameters must have zoned ISO 8061 format i.e. 1970-01-01T00:00:00.000Z</li>
+     * </ul>
+     * @param serverKey key of a server to fetch a message from
+     * @param messageFilters message filters
+     * @param scopes The set (comma separated list of scopes e.g.: STORY, NEXT_BATCHES) of variable's scope<br>
+     * <i>Available scopes:</i>
+     * <ul>
+     * <li><b>STEP</b> - the variable will be available only within the step,
+     * <li><b>SCENARIO</b> - the variable will be available only within the scenario,
+     * <li><b>STORY</b> - the variable will be available within the whole story,
+     * <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     * </ul>
+     * @param variableName A name of variable to assign the values from command execution result
+     * @throws MessagingException signals that an exception of some sort has occurred while communicating with
+     * message service
+     * @throws EmailFetchServiceException signals that an exception of some sort has occurred while fetching
+     * message from email server
+     */
+    @When("I fetch email message from `$serverKey` server filtered by $filters and save message content to $scopes "
+            + "variable `$variableName`")
+    public void saveMessageContent(String serverKey, List<CheckedPredicate<Message, MessagingException>> messageFilters,
+            Set<VariableScope> scopes, String variableName) throws EmailFetchServiceException, MessagingException
+    {
+        EmailServerConfiguration config = serverConfigurations.get(serverKey);
+        List<EmailMessage> messages = messageFetchService.fetch(messageFilters, config);
+
+        int size = messages.size();
+        if (messages.size() != 1)
+        {
+            if (size == 0)
+            {
+                softAssert.recordFailedAssertion("No messages were found by the specified filters");
+                return;
+            }
+            softAssert.recordFailedAssertion(
+                    String.format("Expected one message, but found %d:%n%s", size, formatMessages(messages)));
+            return;
+        }
+
+        EmailMessage message = messages.get(0);
+        List<EmailMessageTextContent> contents = message.getContents();
+        if (softAssert.assertThat("Email text content to save", contents, hasSize(greaterThan(0))))
+        {
+            IntStream.range(0, contents.size()).forEach(index ->
+            {
+                EmailMessageTextContent content = contents.get(index);
+                LOGGER.info("Content {} by index '{}'", content.getContentType(), index);
```

Comment:
atInfo?

Reply:
this is a controversial fix because we do not have any business logic to get args, so it might not be a performance win